src/model_scoring_functions.R

```diff
@@ -0,0 +1,219 @@
+#' Functions to calculate energy scores for model output files according to given reference date
+#'
+#' @param hub_path character string, path to the root of the hub from the current working directory,
+#' defaults to assume that variant-nowcast-hub/src is the working directory
+#' @param model_output_file character string, directory under variant-nowcast-hub/[team name]/[model output parquet file]
+#' @param ref_date character string, date corresponding to model submission deadline, also called nowcast date
+#'
+#' @examples
+#' df_score <- get_energy_scores(hub_path = "../", model_output_file = "UMass-HMLR/2024-10-16-UMass-HMLR.parquet", 
+#' ref_date = "2024-10-16")
+get_energy_scores <- function(
+    hub_path = "../",
+    model_output_file = NULL,
+    ref_date = NULL
+){
+  require("tidyr")
+  require("dplyr")
+  require("readr")
+  require("arrow")
+  require("scoringRules")
+  
+  data <- process_target_data(hub_path = hub_path,
+                          model_output_file = model_output_file,
+                          ref_date = as.Date(ref_date))
+  
+  df_scores <- calc_energy_scores(targets = as.data.frame(data[1]), 
+                                  df_model_output = as.data.frame(data[2]))
```

Comment:
Minor note:

If the output of `process_target_data` is a list of two data frames, then there's no reason to convert the lists to data frames:

```suggestion
  df_scores <- calc_energy_scores(targets = data[[1]], 
                                  df_model_output = data[[2]])
```

Reply:
Yeah, I thought so too! But on my local device, it made `targets` and `df_model_output` have `NULL` values for some reason. I'll see if it was a fluke and try it again. Thanks!

Reply:
Maybe a `[[ ]]`?

Reply:
Ah, I see now. A double bracket `[[ ]]` will resolve that. Thanks, Zhian!